File: test_feature/planning/impl/src/main/java/com/github/klee0kai/stone/test_feature/planning/di/PlanningComponent.java
package com.github.klee0kai.stone.test_feature.planning.di;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.stone.annotations.component.Init;
import com.github.klee0kai.stone.closed.IPrivateComponent;
import com.github.klee0kai.stone.test_feature.planning.PlanningRun;

@Component
public interface PlanningComponent extends IPrivateComponent {

    PlanningDependencies dependencies();

    ProjectsModule projectsModule();

    void inject(PlanningRun planningRun);

    @Init
    void initDep(PlanningDependencies planningRun);

}

File: test_feature/planning/impl/src/main/java/com/github/klee0kai/stone/test_feature/planning/di/PlanningComponent.kt
package com.github.klee0kai.stone.test_feature.planning.di

import com.github.klee0kai.stone.annotations.component.Component
import com.github.klee0kai.stone.annotations.component.Init
import com.github.klee0kai.stone.test_feature.planning.PlanningRun

@Component
interface PlanningComponent {
    fun dependencies(): PlanningDependencies?

    fun projectsModule(): ProjectsModule?

    fun inject(planningRun: PlanningRun?)

    @Init
    fun initDep(planningRun: PlanningDependencies?)
}
